### ISSUE TITLE
Need an example for fields with field type map in the  Value Schema Load generator config

### ISSUE THREAD

**Issue by mark-landry-appD** [corunet/kloadgen#102]
Hi,

The topic I want to load has a number of fields with field type as map.  For example:
field name.                         field type 
payload.entity.details            map

and the failure I get when trying to add a field list value in the value schema load generator config is the following:

org.apache.kafka.common.errors.SerializationException: Error serializing Avro message
Caused by: java.lang.ClassCastException: class java.lang.String cannot be cast to class java.util.Map (java.lang.String and java.util.Map are in module java.base of loader 'bootstrap') in field details in field entity in field payload.

I've tried a few variants field value list values including key1:value1, key2:value2 and I am still facing the same issue.  

What would be the minimum I can provide as a field values list in order to produce to this field type. Also is the Field Length 
 to be used for a map field type?

Thanks in advance

**Comment by kszosze** [comment]
Hi @mark-landry-appD,
I'll take a look into it. Can you please provide a jmx file with the failure you are getting?

Cheers

**Comment by mark-landry-appD** [comment]
Thank you. Will share a version of the jmx shortly.

**Comment by mark-landry-appD** [comment]
jmx attached. Let me know if this is enough information to reproduce. 
[kloadgen_map_example.jmx.txt](https://github.com/corunet/kloadgen/files/6546557/kloadgen_map_example.jmx.txt)

**Comment by kszosze** [comment]
HI @mark-landry-appD ,

I will need the AVRO schema. The map type is wrong, it should be 'string-map' instead of simple map. So probably there is a problem with the import process.
By default if no type is found it will generate a String value.

Cheers.

**Comment by mark-landry-appD** [comment]
Hi @kszosze, 

Here is the portion of the avro schema that is of concern:
    {
      "name": "details",
      "type": {
        "type": "map",
        "values": {
          "type": "array",
          "items": "aaa.bbb.entity.events.avro.AdditionalDetails"
        }
      }
    }





cat AdditionalDetails.avrc
{
  "namespace": "aaa.bbb.entity.events.avro",
  "type": "record",
  "name": "AdditionalDetails",
  "fields": [
    {
      "name": "addType",
      "type": "aaa.bbb.entity.events.avro.AddType"
    },
    {
      "name": "addAmount",
      "type": {
        "type": "bytes",
        "logicalType": "decimal",
        "precision": 15,
        "scale": 5
      }
    },
    {
      "name": "addCode",
      "type": [
        "null",
        "string"
      ],
      "default": null
    },
    {
      "name": "metadata",
      "type": [
        "null",
        {
          "type": "map",
          "values": "string"
        }
      ],
      "default": null
    }
  ]
}


cat AddType.avsc 
{
  "namespace": "aaa.bbb.entity.events.avro",
  "type": "record",
  "name": "AddType",
  "fields": [
    {
      "name": "addTypeId",
      "type": "string"
    },
    {
      "name": "name",
      "type": "string"
    },
    {
      "name": "otherField",
      "type": [
        "null",
        "string"
      ],
      "default": null
    }
  ]
}


Thanks again.

**Comment by kszosze** [comment]
HI @mark-landry-appD,
are you trying to build a Record Array map?
For now, we don´t support such complex structure.
I´ll need to make some development about that. 
I´ll keep you posted since those days are quite busy.
Cheers!

**Comment by mark-landry-appD** [comment]
Hi @kszosze. Thanks for the update.  What would be the target date for release 4.0.0?

**Comment by kszosze** [comment]
Hi @mark-landry-appD ,
Release 4.0.0 will take a bit more than expected since we are working in the Protobuff integration and add a Consumer Sampler to complete the cycle. 
But I can take a look into the schema process and tweak that for you. 
I'll build a sample avro with those pieces you sent to me, or if you wish to provide a better one your be welcome, so I can speed the things.
Cheers

**Comment by mark-landry-appD** [comment]
hi @kszosze, Thanks for having a look at this. We noticed there were a couple of commits and are excited to try it out.   FYI when I tried using the latest commit,  I ran into an issue with a missing class. Exception was  ClassNotFoundException for org.glassfish.jersey.internal.RuntimeDelegateImpl. when I try to load a subject using the value Schema load generator config. I thought maybe this is might be related to some of the changes introduced in the POM related to minimizeJar.
